drivers/serial/uart_gecko.c

```diff
@@ -121,6 +121,11 @@ but not supported by this SOC"
 #error "Driver not supporting hardware flow control for this SOC"
 #endif
 
+#define GECKO_UART_DEFAULT_BAUDRATE	115200
+#define GECKO_UART_DEFAULT_PARITY	UART_CFG_PARITY_NONE
+#define GECKO_UART_DEFAULT_STOP_BITS	UART_CFG_STOP_BITS_1
+#define GECKO_UART_DEFAULT_DATA_BITS	UART_CFG_DATA_BITS_8
```

Comment:
Wouldn't it be better to set these in the DT binding instead? That said, I'm not sure it's a good idea at all to set defaults for these settings, rather it'd be better to have them explicitly in DTS, so that one can easily see what the used settings are. Except for stm32, I don't see any other UART drivers doing this either. Anyway, my order of preference would be:

1. No defaults
2. Put the defaults in the binding (so that the values can be seen in the final `zephyr.dts`)
3. What you have now

Reply:
Actually, it's done in all drivers that attempt to read the values from the DTS (altera/esp32/ifx_cat1/intel_lw/max32), They just don't have the consts at the top of the file. Setting defaults in the DTS would be tough because different drivers would probably have different defaults, don't you think?

Reply:
> Setting defaults in the bindings would be tough because different drivers would probably have different defaults, don't you think?

Every driver has its own binding file. There you can set the default for the properties that the binding inherits from e.g. `uart-controller.yaml`, e.g. like this:
```patch
--- a/dts/bindings/serial/silabs,gecko-uart.yaml
+++ b/dts/bindings/serial/silabs,gecko-uart.yaml
@@ -28,3 +28,6 @@ properties:
     type: array
     required: true
     description: TX pin configuration defined as <location port pin>
+
+  parity:
+    default: "none"
```

Reply:
> Actually, it's done in all drivers that attempt to read the values from the DTS (altera/esp32/ifx_cat1/intel_lw/max32), They just don't have the consts at the top of the file.

Thanks for pointing that out. I indeed missed it since I was looking for `*_DEFAULT` style defines. That said, I still stand by my point that I believe this makes it harder to discover that actual used settings are in the absence of explicit DTS properties, and hence it'd be better to use either per-driver binding defaults or require these explicitly in DTS (which seems like what https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#rules-for-default-values would recommend in this case).

Reply:
Do you want to maybe chime in on the PR (https://github.com/zephyrproject-rtos/zephyr/pull/83324) and try to agree on how this should be so that we don't do things too differently from the rest of the drivers?

Reply:
I'm not sure we're going to "win" the argument but for now maybe we can merge this and continue the discussion in the other PR and work from there?